usr/src/uts/aarch64/promif/prom_init.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License, Version 1.0 only
+ * (the "License").  You may not use this file except in compliance
+ * with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+/*
+ * Copyright 2005 Sun Microsystems, Inc.  All rights reserved.
+ * Use is subject to license terms.
+ */
+/*
+ * Copyright 2025 Michael van der Westhuizen
+ */
+
+#include <sys/promif.h>
+#ifndef _KMDB
+#include <sys/promimpl.h>
+#include <sys/prom_emul.h>
+#include <sys/bootconf.h>
+#include <sys/obpdefs.h>
+#include <sys/kmem.h>
+#endif
+
+int	promif_debug = 0;	/* debug */
+
+/*
+ *  Every standalone that wants to use this library must call
+ *  prom_init() before any of the other routines can be called.
+ */
+/*ARGSUSED*/
+void
+prom_init(char *pgmname, void *cookie)
+{
+#ifndef _KMDB
+	promif_init(pgmname, cookie);
+#endif
+}
+
+#ifndef _KMDB
+/*
```

Comment:
Not sure how to adapt this comment.

The 2.6/10 reference to realmode drivers is x86 specific.
We don't exactly fake a prom tree for /dev/openprom in the FDT world, we just redirect the interfaces to talk to the FDT.

I'm guessing we want to say something about how we either look up the FDT or fake it from ACPI?

Reply:
Yeah, the meat of the comment should probably move to the `promif_setup` for the SBBR kernel.  The FDT implementation is a no-op for the reasons you outline here.